optuna/storages/_in_memory.py

```diff
@@ -323,17 +296,38 @@ def get_trial_param(self, trial_id: int, param_name: str) -> float:
             distribution = trial.distributions[param_name]
             return distribution.to_internal_repr(trial.params[param_name])
 
-    def set_trial_values(self, trial_id: int, values: Sequence[float]) -> None:
+    def set_trial_state_values(
+        self, trial_id: int, state: TrialState, values: Optional[Sequence[float]] = None
+    ) -> bool:
 
         with self._lock:
             trial = self._get_trial(trial_id)
-            self.check_trial_is_updatable(trial_id, trial.state)
 
             trial = copy.copy(trial)
             self.check_trial_is_updatable(trial_id, trial.state)
 
-            trial.values = values
-            self._set_trial(trial_id, trial)
+            if values is not None:
+                trial.values = values
+
+            if state == TrialState.RUNNING and trial.state != TrialState.WAITING:
+                if values is not None:
+                    self._set_trial(trial_id, trial)
```

Comment:
When `state` is `RUNNING`, `values` should be `None`. These lines can be omitted.

Reply:
Right ! Fixed. I also fixed related tests.

And I removed overwrite test from `test_set_trial_state_values_for_values` because I could not find out the way to update existing trial value. If it is possible please tell me.

```
        # Values can be overwritten.
        storage.set_trial_state_values(trial_id_1, state=trial_1.state, values=(0.2,))
        assert storage.get_trial(trial_id_1).value == 0.2
```